bokeh/io.py

```diff
@@ -439,17 +434,13 @@ def _get_save_args(state, filename, resources, title):
 
     return filename, resources, title
 
-def _save_helper(obj, filename, resources, title, validate):
+def _save_helper(obj, state, filename, resources, title, validate):
     remove_after = False
-    if isinstance(obj, LayoutDOM):
-        if obj.document is None:
-            Document().add_root(obj)
```

Comment:
It's currently implemented via `file_html` which uses the new `_ModelInDocument` code to handle theming. I'm not sure what you'd hope to gain from rewriting.